pkg/operator/storageclasscontroller/vmware.go

```diff
@@ -26,14 +62,248 @@ func (c *StorageClassController) newClient(ctx context.Context, cfg *vsphere.VSp
+	properties := []string{DatastoreInfoProperty, SummaryProperty}
+	err = pc.RetrieveOne(ctx, ds.Reference(), properties, &dsMo)
+	if err != nil {
+		return nil, fmt.Errorf("error getting properties of datastore %s: %v", dsName, err)
+	}
+	return &dsMo, nil
+}
+
+func (v *vCenterAPI) createStoragePolicy(ctx context.Context) (string, error) {
+	found, err := v.checkForExistingPolicy(ctx)
+	if err != nil {
+		return v.policyName, fmt.Errorf("error finding existing policy: %v", err)
+	}
+
+	if found {
+		klog.V(3).Infof("found existing storage policy %s", v.policyName)
+		return v.policyName, nil
+	}
+
+	dsName := v.connection.config.Workspace.DefaultDatastore
+	ds, err := v.getDefaultDatastore(ctx)
+	if err != nil {
+		return v.policyName, fmt.Errorf("error fetching default datastore %s: %v", dsName, err)
+	}
+	err = v.createOrUpdateTag(ctx, ds)
+	if err != nil {
+		return v.policyName, fmt.Errorf("error creating or applying tag %s: %v", v.tagName, err)
+	}
+
+	err = v.createStorageProfile(ctx)
+	if err != nil {
+		return v.policyName, fmt.Errorf("error create storage policy profile %s: %v", v.policyName, err)
+	}
+
+	return v.policyName, nil
+}
+
+func (v *vCenterAPI) createOrUpdateTag(ctx context.Context, ds *mo.Datastore) error {
+	// create tag manager for managing tags
+	tagManager := tags.NewManager(v.connection.restClient)
+
+	category, err := tagManager.GetCategory(ctx, v.categoryName)
+	if err != nil && !notFoundError(err) {
+		return fmt.Errorf("error finding category: %+v", err)
+	}
+
+	associatedTypes := appendPrefix(associatedTypesRaw)
+	if category == nil || category.ID == "" {
+		category = &tags.Category{
+			Name:            v.categoryName,
+			Description:     "Container Orchestrator that uses this datastore",
+			AssociableTypes: associatedTypes,
+			Cardinality:     "SINGLE",
+		}
+		catId, err := tagManager.CreateCategory(ctx, category)
+		if err != nil {
+			return fmt.Errorf("error creating category %s: %v", v.categoryName, err)
+		}
+		category.ID = catId
+	} else {
+		existingAssociatedTypes := category.AssociableTypes
+		associatedTypes = updateAssociatedTypes(existingAssociatedTypes)
+		klog.Infof("existing categories are: %+v", existingAssociatedTypes)
+		category.AssociableTypes = associatedTypes
+		klog.Infof("final categories are: %+v", associatedTypes)
+		err := tagManager.UpdateCategory(ctx, category)
+		if err != nil {
+			return fmt.Errorf("error updating category %s: %v", v.categoryName, err)
+		}
+	}
+
+	tag, err := tagManager.GetTag(ctx, v.tagName)
+	if err != nil && !notFoundError(err) {
+		return fmt.Errorf("error finding tag %s: %v", v.tagName, err)
+	}
+	if tag == nil || tag.ID == "" {
+		tag = &tags.Tag{
+			Name:        v.tagName,
+			Description: "Datastore is used by openshift",
+			CategoryID:  category.ID,
+		}
+		tagID, err := tagManager.CreateTag(ctx, tag)
+		if err != nil {
+			return fmt.Errorf("error creating tag %s: %v", v.tagName, err)
+		}
+		tag.ID = tagID
```

Comment:
Please log that a tag has been created.
Again, should this tag be created by the installer? A warning would be appropriate if so.